src/test/java/org/folio/edge/sip2/utils/UtilsTests.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.edge.sip2.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+public class UtilsTests {
+
+  @Test
+  public void testParseQueryString() {
+
+    Map<String, String> queryStringParams = new LinkedHashMap();
+    queryStringParams.put("key1", "value1");
+    queryStringParams.put("key2", "");
+    queryStringParams.put("key3", "value3");
+
+    final String delimeter = " AND ";
+    final String equalDelimeter = "==";
+
+    String expectedString = "key1==value1 AND key3==value3";
+
+    String output = Utils.buildQueryString(queryStringParams, delimeter, equalDelimeter);
+    assertEquals(expectedString, output);
+  }
+
+  @Test
+  public void testParseQueryStringWithEmptyFirstKeyAndAmpercentSeparator() {
+
+    Map<String, String> queryStringParams = new LinkedHashMap();
```

Comment:
This produces a warning in Eclipse, unchecked conversion.
```suggestion
    Map<String, String> queryStringParams = new LinkedHashMap<>();
```